src/containers/course-section/resource-item/ResourceItem.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC, Dispatch, SetStateAction } from 'react'
+import Box from '@mui/material/Box'
+import IconButton from '@mui/material/IconButton'
+import CloseIcon from '@mui/icons-material/Close'
+
+import IconExtensionWithTitle from '~/components/icon-extension-with-title/IconExtensionWithTitle'
+
+import { resourcesData } from '~/containers/course-section/CourseSectionContainer.constants'
+
+import { styles } from '~/containers/course-section/resource-item/ResourceItem.styles'
+
+import { ResourcesTabsEnum as ResourcesTypes, CourseResources } from '~/types'
+
+interface ResourceItemProps {
+  resource: CourseResources
+  setItemToDelete: Dispatch<SetStateAction<CourseResources | null>>
+}
+const ResourceItem: FC<ResourceItemProps> = ({ resource, setItemToDelete }) => {
+  const onDeleteResource = () => {
+    setItemToDelete(resource)
+  }
+
+  const setResourceIcon = () => {
+    if (resource.resourceType === ResourcesTypes.Lessons) {
+      return resourcesData.lesson.icon
+    } else if (resource.resourceType === ResourcesTypes.Quizzes) {
+      return resourcesData.quiz.icon
+    } else {
+      return
+    }
+  }
+  const resourceIcon = setResourceIcon()
```

Comment:
Try refactor function above like:

const setResourceIcon = () => {
  return resource.resourceType in resourcesData ? resourcesData[resource.resourceType].icon : null
}

Also, you can pass setRousceIcon() as icon prop and don`t create resourceIcon variable

Reply:
I have refactored the function a little bit, but it cannot be written as in the example, because of TypeScript errors. In addition, in resourcesData there is an attachment icon for the menu, which should not be used when resources are displayed